+nosnoc/+discrete_time_problem/Cls.m

```diff
@@ -0,0 +1,1600 @@
+classdef Cls < vdx.problems.Mpcc
+    properties
+        model
+        dcs
+        opts
+
+        populated
+    end
+
+    properties(Access=private)
+        z_alg % vdx.VariableGroup:
+        z_impulse % vdx.VariableGroup:
+        z_alg_f_x % vdx.VariableGroup:
+    end
+
+    methods
+        function obj = Cls(dcs, opts)
+            obj = obj@vdx.problems.Mpcc();
+            obj.model = dcs.model;
+            obj.dcs = dcs;
+            obj.opts = opts;
+        end
+
+        function create_variables(obj)
+            dims = obj.dcs.dims;
+            dcs = obj.dcs;
+            model = obj.model;
+            opts = obj.opts;
+
+            % Parameters
+            obj.p.rho_h_p = {{'rho_h_p',1}, 1};
+            obj.p.rho_terminal_p = {{'rho_terminal_p',1}, 1};
+            obj.p.T = {{'T',1}, opts.T};
+            obj.p.p_global = {model.p_global, model.p_global_val};
+
+
+            % 0d vars: Variables which only exist once globally. 
+            obj.w.v_global = {{'v_global',dims.n_v_global}, model.lbv_global, model.ubv_global, model.v0_global};
+            if opts.use_speed_of_time_variables && ~opts.local_speed_of_time_variable
+                obj.w.sot = {{'sot', 1}, opts.s_sot_min, opts.s_sot_max, opts.s_sot0};
+            end
+            if opts.time_optimal_problem
+                obj.w.T_final = {{'T_final', 1}, opts.T_final_min, opts.T_final_max, opts.T};
+                obj.f = obj.f + obj.w.T_final();
+            end
+
+            % 1d vars: Variables that are defined per control stage
+            obj.w.u(1:opts.N_stages) = {{'u', dims.n_u}, model.lbu, model.ubu, model.u0};
+            obj.p.p_time_var(1:opts.N_stages) = {{'p_time_var', dims.n_p_time_var}, model.p_time_var_val};
+            if opts.use_speed_of_time_variables && opts.local_speed_of_time_variable
+                obj.w.sot(1:opts.N_stages) = {{'sot', 1}, opts.s_sot_min, opts.s_sot_max, opts.s_sot0};
+            end
+
+            % TODO(@anton) This _severely_ hurts performance over the vectorized assignment by doing N_stages vertcats of
```

Comment:
we have this comment in all discrete_time_problem classes, but this is more of a vdx issue?

Reply:
The details yea are perhaps a VDX issue but it would be good to keep a note here for reference regarding performance. I plan on fixing the performance issue in VDX next week and we can remove this.